.github/workflows/monthly-builds.yaml

```diff
@@ -0,0 +1,118 @@
+      # URL to this repo echoed below to help debug builds (does not change if the workflow is re-run)
+      SNAPCRAFT_BUILDER_ID: ${{ github.run_id }}
+    steps:
+      - name: Download snapcrafts folders
+        uses: actions/download-artifact@v3
+        with:
+          name: snaps
+      - name: print matrix
+        run: |
+          echo ${{ matrix.snap_names }}
+      - name: Add LP credentials
+        run: |
+          mkdir -p ~/.local/share/snapcraft/provider/launchpad/
+          echo '${{ secrets.LP_CREDS }}' > ~/.local/share/snapcraft/provider/launchpad/credentials
+          git config --global user.email "canonical-robotics-brand@canonical.com"
+          git config --global user.name "Canonical robotics"
+      - name: Print Launchpad build repository
+        run: |
+          echo "Building at: https://git.launchpad.net/~ce-certification-qa/+snap/$SNAPCRAFT_BUILDER_ID"
+      - uses: snapcore/action-build@v1
+        with:
+          path: /home/runner/work/ros-content-sharing-snaps/ros-content-sharing-snaps/${{ matrix.snap_names }}
+          snapcraft-args: "remote-build --launchpad-accept-public-upload --build-id ${{ matrix.snap_names }}-${SNAPCRAFT_BUILDER_ID}"
+      - name: check number of built snaps
+        run: |
+          count_txt_files=$(find /home/runner/work/ros-content-sharing-snaps/ros-content-sharing-snaps/${{ matrix.snap_names }} -type f -name 'ros-*.txt' | wc -l)
+          count_snap_files=$(find ${{ matrix.snap_names }} -type f -name '*.snap' | wc -l)
+          if [ "$count_txt_files" -ne "$count_snap_files" ]; then
+            echo "Error: The number of '.txt' files and '.snap' files do not match."
+            exit 1
+          fi
+      - uses: actions/upload-artifact@v3
+        name: Upload snapcraft logs
+        if: always()
+        with:
+          name: snapcraft_logs
+          path: |
+            /home/runner/.cache/snapcraft/log/
+            /home/runner/.local/state/snapcraft/log/
+            /home/runner/work/ros-content-sharing-snaps/ros-content-sharing-snaps/**/ros-*.txt
+      - uses: actions/upload-artifact@v3
+        name: Upload the snap as artifact
+        with:
+          name: ${{ matrix.snap_names }}
+          path: /home/runner/work/ros-content-sharing-snaps/ros-content-sharing-snaps/${{ matrix.snap_names }}/*.snap
+
+  ros-publish:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: [generate-snapcrafts, snap-build]
+    strategy:
+      fail-fast: false
+      matrix:
+        snap_names: ${{ fromJSON(needs.generate-snapcrafts.outputs.matrix) }}
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: ${{ matrix.snap_names }}
+        path: .
+    - name: Get snap filename
+      run: echo "SNAPFILE=$(ls *.snap)" >> $GITHUB_ENV
+    - uses: snapcore/action-publish@v1
+      env:
+        SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+      with:
+        snap: ${{ env.SNAPFILE }}
+        release: edge
```

Comment:
so here we are releasing on edge and then manually moving them to stable?

Reply:
I guess this is to discuss.

Reply:
After discussion:
- we push on edge
- we must add a workflow to release from edge to stable